sanic/response.py

```diff
@@ -319,9 +319,26 @@ def html(
     )
 
 
+async def validate_file(
+    request_headers: Header, last_modified: Union[datetime, float, int]
+):
+    if_modified_since = request_headers.get("If-Modified-Since")
```

Comment:
What about `If-None-Match` support?

Reply:
We don't have eTag for both static files and `file` response function at this time. So I would suggest doing this in another PR. By the way, may I try to combine the logics that handles validating files for static files and the `file` response function in order to reduce duplicated functionality of code?